### PR TITLE
Refine length prefix utilities

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- simplify `bytes_to_u64` and `u64_to_bytes`
- cover max and zero-length cases in frame tests

## Testing
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_688bc1a7a4e0832289453fc8d44c9c7e

## Summary by Sourcery

Simplify length prefix conversion utilities and enhance their test coverage

Enhancements:
- Remove the custom PrefixErr enum and parse_bytes helper in favor of inline error creation
- Refactor bytes_to_u64 to use a fixed 8-byte buffer and inline endianness handling
- Refactor u64_to_bytes to construct prefix bytes directly and inline size validation

Tests:
- Add tests for maximum prefix values (all-0xFF cases) for bytes_to_u64
- Add a test to ensure u64_to_bytes fails on zero-length prefixes

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- In u64_to_bytes, explicitly zero out out[size..] (as was done previously) to avoid leaving stale data in the buffer.
- Rather than allocating a Vec for prefix_bytes, consider writing directly into the out slice per size/endianness to avoid heap allocations.

File: src/frame/conversion.rs [hunk 1]
```diff
@@ -74,38 +56,57 @@ pub fn bytes_to_u64(bytes: &[u8], size: usize, endianness: Endianness) -> io::Re
 /// # Errors
 /// Returns [`io::ErrorKind::InvalidInput`] if the size is unsupported or if
 /// `len` does not fit into the prefix.
+///
+/// # Panics
+/// Panics if shifting `value` leaves bits outside the `u8` range. This cannot
+/// occur for valid prefix sizes and checked values.
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/frame/conversion.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai review
